### ISSUE TITLE
Add Git commit checks to disallow `.unwrap()` 

### ISSUE THREAD

**Issue by tvolk131** [fedimint/fedimint#4313]
Are there any cases where we'd prefer `.unwrap()` over `.expect()`? If not, I think adding a pre-commit hook check to ensure `.unwrap()` is not used could speed up PR review for newcomers.

**Comment by elsirion** [comment]
In tests I considered `unwrap`s good enough in the past, but I'd be ok with only allowing `expect`s going forward.

**Comment by elsirion** [comment]
Can we limit the check to newly introduced code @dpc?

**Comment by dpc** [comment]
Nope. I'm not aware of good way to run stuff "for new code only". Changes like this require fixing the code fix, then trying to enforce it.

**Comment by tvolk131** [comment]
Could we enforce this within certain subdirectories? That'd allow for us to migrate to no-unwrap incrementally, enforcing each subdirectory as we go to prevent regressions. This might be a pain to manage but just want to put the idea out there.

**Comment by kernelkind** [comment]
> Could we enforce this within certain subdirectories? 

Yep, super easy with `semgrep`
